include/wx/msw/ole/oleutils.h

```diff
@@ -65,24 +65,26 @@ inline void wxOleUninitialize()
 
 class WXDLLIMPEXP_CORE wxBasicString
 {
-public:    
-    // Takes over the ownership of bstr
-    explicit wxBasicString(BSTR bstr = NULL);
+public:
+    // Constructs with the owned BSTR set to NULL
+    wxBasicString() : m_bstrBuf(NULL) {}
 
-    // Constructs the BSTR from wxString
-    wxBasicString(const wxString& str);
+    // Constructs with the owned BSTR created from a wxString
+    wxBasicString(const wxString& str)
+        : m_bstrBuf(SysAllocString(str.wc_str(*wxConvCurrent))) {}
 
-    // Creates a copy of the BSTR owned by bstr
-    wxBasicString(const wxBasicString& bstr);
+    // Constructs the owned BSTR as a copy of the BSTR owned by bstr
+    wxBasicString(const wxBasicString& bstr) : m_bstrBuf(bstr.Copy()) {}
 
     // Frees the owned BSTR
-    ~wxBasicString();
+    ~wxBasicString() { SysFreeString(m_bstrBuf); }
 
-    // Returns the owned BSTR and gives up its ownership
-    BSTR Detach();
+    // Creates the owned BSTR from a wxString
+    void AssignFromwxString(const wxString& str);
```

Comment:
Could we call it just `AssignFromString()`? This reads better and it does take any string due to implicit conversions to `wxString`, after all.